meta-lmp-bsp/recipes-security/optee/optee-os-fio-bsp-common.inc

```diff
@@ -3,6 +3,28 @@ OPTEE_HEADER:stm32mp1common    = "tee-header_v2"
 OPTEE_PAGEABLE:stm32mp1common  = "tee-pageable_v2"
 OPTEE_PAGER:stm32mp1common     = "tee-pager_v2"
 
+# TI HS signing logic (from meta-ti-bsp optee-os bbappend)
+do_compile:prepend:k3() {
+    export TI_SECURE_DEV_PKG=${TI_SECURE_DEV_PKG}
+}
+# Signing procedure for K3 HS devices
+optee_sign_k3hs() {
+    ( cd ${B}/core/; \
+        if [ -f ${TI_SECURE_DEV_PKG}/scripts/secure-binary-image.sh ]; then \
+            ${TI_SECURE_DEV_PKG}/scripts/secure-binary-image.sh tee-pager_v2.bin tee-pager_v2.bin.signed; \
+        else \
+            echo "Warning: TI_SECURE_DEV_PKG not set, OP-TEE not signed."; \
+            cp tee-pager_v2.bin tee-pager_v2.bin.signed; \
+        fi; \
+    )
+}
+do_compile:append:k3() {
+    optee_sign_k3hs
+}
+do_install:append:k3() {
+    install -m 644 ${B}/core/tee-pager_v2.bin.signed ${D}${nonarch_base_libdir}/firmware/
```

Comment:
At the end maybe we can only use this install that does the rename at same time
```
install -m 644 ${B}/core/tee-pager_v2.bin ${D}${nonarch_base_libdir}/firmware/tee-pager_v2.bin.signed
```

Reply:
For non signed cases it will be the same binary, but when signing is enabled we need to copy tee-pager_v2.bin.signed instead.

Reply:
with the NDA now makes sense.